systemd/sdnotify.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package systemd
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/logger"
+)
+
+func RunWatchdog() (*time.Ticker, error) {
+	wu := os.Getenv("WATCHDOG_USEC")
+	if wu == "" {
+		return nil, fmt.Errorf("cannot get WATCHDOG_USEC environment")
+	}
+	usec, err := strconv.Atoi(wu)
```

Comment:
Not sure it's wanted/helpful for this case, but note you have `osutil.GetenvInt64`.